e2e/app/solvernetrouting.go

```diff
@@ -0,0 +1,148 @@
+package app
+
+import (
+	"context"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/e2e/app/eoa"
+	"github.com/omni-network/omni/lib/contracts"
+	"github.com/omni-network/omni/lib/contracts/solvernet"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient/ethbackend"
+	"github.com/omni-network/omni/lib/netconf"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// SetSolverNetRoutes sets the SolverNet routes for all chains in a given network.
+func SetSolverNetRoutes(ctx context.Context, network netconf.Network, backends ethbackend.Backends) error {
+	addrs, err := contracts.GetAddresses(ctx, network.ID)
+	if err != nil {
+		return errors.Wrap(err, "get addresses", "network", network.ID)
+	}
+
+	chainIDs := network.EVMChains()
+	for _, chain := range chainIDs {
+		remoteChainIDs, remoteOutboxes, remoteInboxConfigs, err := getRoutes(chain, chainIDs, addrs.SolverNetInbox, addrs.SolverNetOutbox)
+		if err != nil {
+			return errors.Wrap(err, "get routes", "chain", chain.Name)
+		}
+
+		backend, err := backends.Backend(chain.ID)
+		if err != nil {
+			return errors.Wrap(err, "get backend", "chain", chain.Name)
+		}
+
+		isDeployed, err := isDeployed(ctx, backend, addrs.SolverNetInbox, addrs.SolverNetOutbox)
+		if !isDeployed || err != nil {
+			return errors.Wrap(err, "isDeployed", "chain", chain.Name)
+		}
+
+		txOpts, err := backend.BindOpts(ctx, eoa.MustAddress(network.ID, eoa.RoleManager))
+		if err != nil {
+			return errors.Wrap(err, "bind opts", "chain", chain.Name)
+		}
+
+		err = configureInbox(ctx, backend, txOpts, addrs.SolverNetInbox, remoteChainIDs, remoteOutboxes)
+		if err != nil {
+			return errors.Wrap(err, "configure inbox", "chain", chain.Name)
+		}
+
+		err = configureOutbox(ctx, backend, txOpts, addrs.SolverNetOutbox, remoteChainIDs, remoteInboxConfigs)
+		if err != nil {
+			return errors.Wrap(err, "configure outbox", "chain", chain.Name)
+		}
+	}
```

Comment:
This will take a while to run on each chain. Let's use an `errgroup` (though we can do this after first merge)